x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -901,17 +818,15 @@ private Expression replaceAliases(Expression condition, List<? extends NamedExpr
         }
     }
 
-    private class ResolveFunctions extends AnalyzerRule<LogicalPlan> {
+    private static class ResolveFunctions extends AnalyzerRules.ResolveFunctions {
+
+        ResolveFunctions(Configuration configuration, FunctionRegistry functionRegistry) {
```

Comment:
The configuration is per request while the Rule across requests. This needs to be broken out so `ResolveFunctions` is abstract and each QL project extends it as part of their own Analyzer and thus have access to their configuration.
Moving forward we could parameterized this so the Analyzer (and its rules) become singleton.

Reply:
Reworked after the parameterised rule and executor changes.